src/main/java/com/minecraftabnormals/abnormals_core/core/util/DataUtil.java

```diff
@@ -314,4 +301,89 @@ public static void registerConfigPredicate(IConfigPredicateSerializer<?> seriali
+	 * block is in front of the dispenser, authors may want to make sure that {@code b}'s condition is registered after
+	 * {@code a}'s. In this case, {@code a}'s {@code modComparator} should return -1 if the string given by the function
+	 * equals {@code b}, and {@code b}'s should return 1 if the string equals {@code a}.</p>
+	 *
+	 * <p>Ideally, the condition should be implemented such that the predicate only passes if the new behavior will be 'successful', avoiding problems with failure sounds not playing.</p>
+	 *
+	 * <p>Since Abnormals Core handles registering the condition at the right time, mods should call this method as early as possible,
+	 * such as in a {@code static{}} block.</p>
+	 *
+	 * @param item The {@link Item} to register the {@code newBehavior} for.
+	 * @param condition A {@link BiPredicate} that takes in {@link IBlockSource} and {@link ItemStack} arguments, returning true if the {@code newBehavior} should be performed.
+	 * @param behavior The {@link IDispenseItemBehavior} that will be used if the {@code condition} is met.
+	 * @param modComparator A {@link Function} that inputs a String representing the mod id for another behavior registered to the same item.
+	 *                      It should return 1 if {@code behavior} is to be registered after the other behavior, -1 if
+	 *                      it should go before, and 0 in any other case.
+	 *
+	 * @author abigailfails
+	 */
+	public static void registerAlternativeDispenseBehavior(String modId, Item item, BiPredicate<IBlockSource, ItemStack> condition, IDispenseItemBehavior behavior, Function<String, Integer> modComparator) {
+		ALTERNATIVE_DISPENSE_BEHAVIORS.add(new AlternativeDispenseBehavior(modId, item, condition, behavior, modComparator));
+	}
+
+	public static class AlternativeDispenseBehavior implements Comparable<AlternativeDispenseBehavior> {
+		private final String modId;
+		private final Item item;
+		private final BiPredicate<IBlockSource, ItemStack> condition;
+		private final IDispenseItemBehavior behavior;
+		private final Function<String, Integer> modIdComparator;
+
+		public AlternativeDispenseBehavior(String modId, Item item, BiPredicate<IBlockSource, ItemStack> condition, IDispenseItemBehavior behavior, Function<String, Integer> modIdComparator) {
+			this.modId = modId;
+			this.item = item;
+			this.condition = condition;
+			this.behavior = behavior;
+			this.modIdComparator = modIdComparator;
+		}
+
+		@Override
+		public int compareTo(AlternativeDispenseBehavior behavior) {
+			return this.item == behavior.item ? modIdComparator.apply(behavior.modId) : 0;
+		}
+
+		public void register() {
+			IDispenseItemBehavior oldBehavior = DispenserBlock.DISPENSER_REGISTRY.get(item);
+			DispenserBlock.registerBehavior(item, (source, stack) -> condition.test(source, stack) ? behavior.dispense(source, stack) : oldBehavior.dispense(source, stack));
+		}
+
+		public static List<AlternativeDispenseBehavior> alternativeDispenseBehaviors() {
```

Comment:
Because this is a method that copies and sorts the behaviors, `getSortedAlternativeDispenseBehaviors()` would be a more fitting name.
```suggestion
		public static List<AlternativeDispenseBehavior> getSortedAlternativeDispenseBehaviors() {
```

Reply:
Done in latest commit.